src/admin/config_schema.py

```
@@ -0,0 +1,169 @@
+    all_aliases: list[str] = []
+    all_user_refs: set[str] = set()
+
+    for community in config.communities:
+        for space in community.spaces.values():
+            all_aliases.extend(_collect_aliases(space))
+            all_user_refs.update(_collect_user_refs(space))
+
+    seen = set()
+    for alias in all_aliases:
+        if alias in seen:
+            raise ConfigError(f"Duplicate room alias: {alias}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)